### ISSUE TITLE
config:set fails with JSON values

### ISSUE THREAD

**Issue by MichaelThessel** [magento/magento2#22396]
### Preconditions (*)

1. 2.3.1

### Steps to reproduce (*)

1. Try to set a config value for a config option that uses JSON values via the command line cli
2. i.e.: `bin/magento config:set --scope default cataloginventory/item_options/min_sale_qty '{"_q0efyLTwddknOffVxvtSNpjQYmv9eAe9":{"customer_group_id":32000,"min_sale_qty":1}}' --lock-config`

### Expected result (*)

1. Config is set to provided value

### Actual result (*)

1. Config is set to empty value.
2. In the example above:

```
diff --git a/app/etc/config.php b/app/etc/config.php
index f3da38ee..34f30164 100644
--- a/app/etc/config.php
+++ b/app/etc/config.php
@@ -229,5 +229,15 @@ return [
         'Magento_WishlistGraphQl' => 1,
         'Temando_Shipping' => 1,
         'Vertex_Tax' => 1
+    ],
+    'system' => [
+        'default' => [
+            'cataloginventory' => [
+                'item_options' => [
+                    'min_sale_qty' => '',
+                ]
+            ]
+        ]
     ]
 ];
```

**Comment by kalpmehta** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, here is your [Magento instance](https://i-22396-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22396-2-3-develop.instances.magento-community.engineering/admin](https://i-22396-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by kalpmehta** [comment]
@MichaelThessel The value should be given like below I think but still it's not working for me as well.

`bin/magento config:set --scope default cataloginventory/item_options/min_sale_qty '{"32000":1}' --lock-config`

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @kalpmehta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99255` were created

**Issue Available**: @kalpmehta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by MichaelThessel** [comment]
@kalpmehta what I basically did was to set the value in admin. Then I did a `config:show` and copied the value from there. I would assume that the values in `config:show` and `config:set` should be identical. If that's not the case then I would think this would be another bug.

**Comment by magento-engcom-team** [comment]
Hi @MichaelThessel. Thank you for your report. 
The issue has been fixed in magento/magento2#23277 by @shikhamis11 in 2.2-develop branch
Related commit(s):
 - [6c46ed078f3896071c69704f1b206436924fa2e7](https://github.com/magento/magento2/commit/6c46ed078f3896071c69704f1b206436924fa2e7)

The fix will be available with the upcoming 2.2.10 release.
